### ISSUE TITLE
Make usernames case-insensitive

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#554]
**TL;DR**: Usernames on Lookit should be case-insensitive.

**Narrative**
"As a Participant, I get confused if I registered under e.g. First_last@gmail.com, but then can't log in using my email address first_last@gmail.com. If I request a password reset, I receive it successfully and can reset the password - but then can't actually log in because I've actually reset the password for First_last@gmail.com."

"As a Participant, if I can't log into my account I may figure I didn't really have one and try creating a new account, which may by chance have different capitalization. This can cause errors and it's also confusing to have multiple accounts unintentionally."

We had about 20 duplicate-up-to-case usernames as of several months ago, and I manually cleaned them up on prod (merging/deleting accounts as appropriate). I've handled tech support for 3 participants over 4-5 days (out of ~300) who were having trouble logging in after resetting their password which turned out to be due to the capitalization of their username.

(Note: this affects researchers too, there are just fewer of us and we seem more likely to be using password managers and all-lowercase email addresses for registration.)

**Acceptance Criteria**

- [x] Manual cleanup of staging, prod databases to avoid duplicate accounts
- [x] Migration that safely changes email addresses for duplicate accounts (e.g. adding +<UNIQUESTRING>) so we can fix any that are added right before 
- [x] Add tests for account creation and username changes(don't allow duplicates), for what happens if duplicate account does exist when logging in, and for logging in as user@company.com and UsEr@ComPany.COM 
- [x] QA on new feature: double check we can log in to previously-existing account originally under all-lowercase; can log in to previously-existing account with capital letters; can not change username to an existing address with capitalization changes; can not register new account using existing address with capitalization changes; can send email to participant; can trigger password reset email, reset password, and log in successfully

**Implementation Notes**
Could have migration actually try to handle merging accounts (e.g., move all children to acct 1, all messages/logs/etc. reference acct 1) and then match children manually afterwards, but I think that'll be substantially more trouble than it's worth to test carefully enough, versus just handling the relatively few cases manually.

**Comment by Datamance** [comment]
OK - this can also be future-proofed by overriding `get_by_natural_key` on our custom `UserManager` a la:

```python
def get_by_natural_key(self, username):
        return self.get(**{self.model.USERNAME_FIELD + '__iexact': username})
```

Lets talk on slack about how to remediate this issue otherwise.
